frontend/src/pages/acceleratorProfiles/screens/manage/IdentifierSelectField.tsx

```diff
@@ -0,0 +1,68 @@
+import { Select, SelectOption, TextInput } from '@patternfly/react-core';
+import React, { useEffect, useMemo } from 'react';
+
+type IdentifierSelectFieldProps = {
+  value: string;
+  onChange: (identifier: string) => void;
+  identifierOptions: string[];
+};
+
+export const IdentifierSelectField = ({
+  value,
+  onChange,
+  identifierOptions = [],
+}: IdentifierSelectFieldProps) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  // remove possible duplicates
+  const options = useMemo(() => Array.from(new Set(identifierOptions)), [identifierOptions]);
+
+  // auto-select if there is only one option
+  useEffect(() => {
+    if (options.length === 1) {
+      onChange(options[0]);
+    }
+    // Do not include onChange callback as dependency
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [options]);
+
+  if (options.length > 1) {
+    return (
+      <Select
+        removeFindDomNode
+        id="accelerator-identifier-select"
+        data-testid="accelerator-identifier-select"
+        isOpen={isOpen}
+        placeholderText="Select an identifier"
+        onToggle={(open) => setIsOpen(open)}
+        onSelect={(_, option) => {
+          if (typeof option === 'string') {
+            onChange(option);
+            setIsOpen(false);
+          }
+        }}
+        selections={value}
+      >
+        {options.map((option) => (
+          <SelectOption key={option} value={option}>
+            {option}
+          </SelectOption>
+        ))}
+      </Select>
+    );
+  }
+
+  return (
+    <TextInput
+      isRequired
+      value={value}
+      id="accelerator-identifier"
+      name="accelerator-identifier"
+      isDisabled={options.length === 1}
+      onChange={(identifier) => onChange(identifier)}
+      placeholder="Example, nvidia.com/gpu"
```

Comment:
Do we need placeholder here? As it will always have one option which is disabled.

Reply:
we do, that case it can be empty and not autfilled

Reply:
This would still be needed when the creation form is used standalone without a predefined option.